flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/harness/JoinHarnessTest.scala

```diff
@@ -232,4 +232,164 @@ class JoinHarnessTest extends HarnessTestBase{
     testHarness.close()
   }
 
+  /** a.proctime >= b.proctime - 10 and a.proctime <= b.proctime + 20 **/
+  @Test
+  def testNewNormalProcTimeJoin() {
```

Comment:
We would also need harness tests for the event-time join.
In addition we need:
- plan translation tests (see `org.apache.flink.table.api.stream.sql.JoinTest`)
- some end-to-end ITCases (see `org.apache.flink.table.runtime.stream.sql.JoinITCase` but with result validation)

Moreover, the processing time restriction for windowed joins should be removed from the documentation.